gcloud-java-compute/src/main/java/com/google/gcloud/compute/Operation.java

```diff
@@ -0,0 +1,800 @@
+
+    com.google.api.services.compute.model.Operation.Error.Errors toPb() {
+      return new com.google.api.services.compute.model.Operation.Error.Errors()
+          .setCode(code)
+          .setLocation(location)
+          .setMessage(message);
+    }
+
+    static OperationError fromPb(
+        com.google.api.services.compute.model.Operation.Error.Errors errorPb) {
+      return new OperationError(errorPb.getCode(), errorPb.getLocation(), errorPb.getMessage());
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      return obj instanceof OperationError
+          && Objects.equals(code, ((OperationError) obj).code)
+          && Objects.equals(message, ((OperationError) obj).message)
+          && Objects.equals(location, ((OperationError) obj).location);
+    }
+
+    @Override
+    public String toString() {
+      return MoreObjects.toStringHelper(this)
+          .add("code", code)
+          .add("location", location)
+          .add("message", message)
+          .toString();
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(code, location, message);
+    }
+  }
+
+  /**
+   * A warning message that is generated during the processing of a Google Compute Engine operation.
+   */
+  public static final class OperationWarning implements Serializable {
+
+    static final
+        Function<com.google.api.services.compute.model.Operation.Warnings, OperationWarning>
+        FROM_PB_FUNCTION =
+        new Function<com.google.api.services.compute.model.Operation.Warnings,
+            OperationWarning>() {
+          @Override
+          public OperationWarning apply(
+              com.google.api.services.compute.model.Operation.Warnings pb) {
+            return OperationWarning.fromPb(pb);
+          }
+        };
+    static final
+        Function<OperationWarning, com.google.api.services.compute.model.Operation.Warnings>
+        TO_PB_FUNCTION =
+        new Function<OperationWarning,
+            com.google.api.services.compute.model.Operation.Warnings>() {
+          @Override
+          public com.google.api.services.compute.model.Operation.Warnings apply(
+              OperationWarning operation) {
+            return operation.toPb();
+          }
+        };
+
+    private static final long serialVersionUID = 4917326627380228928L;
+
+    private final String code;
+    private final String message;
+    private final Map<String, String> metadata;
+
+    OperationWarning(String code, String message, Map<String, String> metadata) {
+      this.code = code;
+      this.metadata = metadata != null ? ImmutableMap.copyOf(metadata) : null;
+      this.message = message;
+    }
+
+    /**
+     * Returns a warning identifier for this warning.
+     */
+    public String code() {
+      return code;
+    }
+
+    /**
+     * Returns a human-readable error message.
+     */
+    public String message() {
+      return message;
+    }
+
+    /**
+     * Returns metadata about this warning.
```

Comment:
Do you know if there's any docs/consistency on what kind of metadata is provided?  If so, it'd be good to add to the javadoc.